### PR TITLE
Fix node naming in {param,field}_declaration

### PR DESCRIPTION
**Pull request by adonovan**
This change commutes the field(...) and commaSep(...) operations so that the commas don't get named.
The set of accepted inputs has not changed.

Fixes https://github.com/tree-sitter/tree-sitter-go/issues/45

Checklist:
- [ ] All tests pass in CI.
- [ ] There are sufficient tests for the new fix/feature.
- [ ] Grammar rules have not been renamed unless absolutely necessary.
- [ ] The conflicts section hasn't grown too much.
- [ ] The parser size hasn't grown too much (check the value of STATE_COUNT in src/parser.c).

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -232,7 +232,8 @@ module.exports = grammar({
     ),
 
     parameter_declaration: $ => seq(
-      field('name', commaSep($.identifier)),
+      field('name', $._identifier),
+      repeat(seq(',', field('name', $._identifier))),
```
Comment (maxbrunsfeld) [inline grammar.js:236]: I think that you could still use the `commaSep` function for this:

```js
commaSep(field('name', $.identifier)),
```

**Review by maxbrunsfeld — COMMENTED** [review]
Thanks for fixing this. I left one stylistic suggestion.

File: grammar.js (hunk 1 above)
Comment (adonovan) [reply grammar.js]: Of course; done.

**Comment by maxbrunsfeld** [comment]
Can you update the generated parser by running `tree-sitter generate`? We currently check the generated code into these repositories.

**Comment by adonovan** [comment]
> Can you update the generated parser by running `tree-sitter generate`? We currently check the generated code into these repositories.

Done. Is there a document that explains the usual contributing process that I can add a link to from this README?

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
This need to be rebased from the latest.
Sorry for accepting this a bit later than your other PRs.

**Pull request merged**